solid/src/main/java/com/inrupt/client/solid/SolidProblemDetails.java

```diff
@@ -133,51 +138,29 @@ private static JsonService getJsonService() {
 
     /**
      * Builds a {@link ProblemDetails} instance from an HTTP error response.
+     * @param uri the original URI
      * @param statusCode the HTTP error response status code
      * @param headers the HTTP error response headers
      * @param body the HTTP error response body
      * @return a {@link ProblemDetails} instance
      */
-    public static SolidProblemDetails fromErrorResponse(
-            final int statusCode,
-            final Headers headers,
-            final byte[] body
-    ) {
+    static SolidProblemDetails fromErrorResponse(final URI uri, final int statusCode, final Headers headers,
+            final byte[] body) {
         final JsonService jsonService = getJsonService();
-        if (jsonService == null
-                || (headers != null && !headers.allValues("Content-Type").contains(ProblemDetails.MIME_TYPE))) {
-            return new SolidProblemDetails(
-                URI.create(ProblemDetails.DEFAULT_TYPE),
-                null,
-                null,
-                statusCode,
-                null
-            );
-        }
-        try {
-            final Data pdData = jsonService.fromJson(
-                    new ByteArrayInputStream(body),
-                    Data.class
-            );
-            final URI type = Optional.ofNullable(pdData.type)
-                .orElse(URI.create(ProblemDetails.DEFAULT_TYPE));
-            // JSON mappers map invalid integers to 0, which is an invalid value in our case anyway.
-            final int status = Optional.of(pdData.status).filter(s -> s != 0).orElse(statusCode);
-            return new SolidProblemDetails(
-                    type,
-                    pdData.title,
-                    pdData.details,
-                    status,
-                    pdData.instance
-            );
-        } catch (IOException e) {
-            return new SolidProblemDetails(
-                URI.create(ProblemDetails.DEFAULT_TYPE),
-                null,
-                null,
-                statusCode,
-                null
-            );
+        if (jsonService != null
+                && (headers != null && headers.allValues("Content-Type").contains(ProblemDetails.MIME_TYPE))) {
+            try (final InputStream input = new ByteArrayInputStream(body)) {
+                final Data pdData = jsonService.fromJson(input, Data.class);
+                final URI type = Optional.ofNullable(pdData.type)
+                    .map(uri::resolve)
```

Comment:
The same is required for the instance URI

Reply:
Addressed in https://github.com/inrupt/solid-client-java/pull/1393/commits/89c65a35adc537e415874b4608963482c2966a92